From ee66e2b0e61a2f592797ddca8cef48b970e5223c Mon Sep 17 00:00:00 2001
From: Jayanth Othayoth <ojayanth@in.ibm.com>
Date: Fri, 28 Jan 2022 04:29:43 -0600
Subject: [PATCH] attribute: disable oscrefclk cronus target support

oscrefclk is BMC only target and it is not modeled by fapi.
This target is not consumed by fapi/cronus now.
Namining convention used by existing pdata attribute is incorrect
and causing attribute import/translate function related oscrefclk
target. Added workaround to fix attributes tool import function
to disable cronus function target support for oscrefclk.

BMC Impact: During the attribute import function, which is used
in concurrent code update path is not going to preserve HWAS_STATE
attribute value, with previous boot configuration. Since oscrefclk
HWAS_STATE attribute is not consumed in MPIPL path( this is the only
use case), no impact in any of the existing BMC functionalities.

tested: verified attributes tool import function

Signed-off-by: Jayanth Othayoth <ojayanth@in.ibm.com>
---
 libdtree/dtree_util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libdtree/dtree_util.c b/libdtree/dtree_util.c
index c3cddf3..8ffee80 100644
--- a/libdtree/dtree_util.c
+++ b/libdtree/dtree_util.c
@@ -77,7 +77,7 @@ struct {
 
 	{ "TARGET_TYPE_TPM", "tpm", "tpm" },
 	{ "TARGET_TYPE_BMC", "bmc", "bmc" },
-	{ "TARGET_TYPE_OSCREFCLK", "oscrefclk", "oscrefclk" },
+	{ "TARGET_TYPE_OSCREFCLK", "oscrefclk", NULL },
 	{ NULL, NULL, NULL },
 };
 
-- 
1.8.3.1

